openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/FairGBMDescriptorUtil.java

```diff
@@ -0,0 +1,218 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Utility to organize all the necessary Machine Learning Hyper-Parameters for configuring the training of LightGBM.
+ *
+ * @author Andre Cruz (andre.cruz@feedzai.com)
+ * @since 1.3.6
+ */
+public class FairGBMDescriptorUtil extends LightGBMDescriptorUtil {
+
+    public static final String CONSTRAINT_GROUP_COLUMN_PARAMETER_NAME = "constraint_group_column";
+
+    /**
+     * Defines the set of model parameters supported by the FairGBM algorithm.
+     */
+    static final Set<ModelParameter> PARAMS = Sets.union(ImmutableSet.of(
+            // The single parameter that will change for every different dataset
+            new ModelParameter(
+                    CONSTRAINT_GROUP_COLUMN_PARAMETER_NAME,
+                    "(Fairness) Sensitive group column",
+                    "Fairness constraints are enforced over this column.\n"
+                            + "This column must be in categorical format.\n"
+                            + "Start this string with `name:` to use the name of a column, \n"
+                            + "e.g., `name:age_group` for a column named `age_group`.",
+                    MANDATORY,
+                    new FreeTextFieldType("")
+//                    new FreeTextFieldType("", ".+")       # TODO: https://github.com/feedzai/feedzai-openml/issues/68
+            ),
+
+            new ModelParameter(
+                    "constraint_type",
+                    "(Fairness) Constraint type",
+                    "Enforces group-wise parity on the given target metric for the selected group column. "
+                            + "In general, FPR can be used for most detection settings "
+                            + "to equalize the negative outcomes on legitimate individuals "
+                            + "(false positives).",
+                    NOT_MANDATORY,
+                    new ChoiceFieldType(ImmutableSet.of("FPR", "FNR", "FPR,FNR"), "FPR")
+            ),
+
+            // Parameters related to global constraints
+            new ModelParameter(
+                    "global_constraint_type",
+                    "(Fairness) Global constraint type",
+                    "FairGBM modifies the output scores to meet your target FPR and/or FNR as well as "
+                            + "fairness at a decision threshold of approximately 0.5 (or 500 in Pulse). Set parameters "
+                            + "(Fairness) Global target FPR/FNR accordingly. Using decision thresholds far from 0.5 "
+                            + "will not ensure fairness.",
+                    NOT_MANDATORY,
+                    new ChoiceFieldType(ImmutableSet.of("FPR", "FNR", "FPR,FNR"), "FPR,FNR")
+            ),
+            new ModelParameter(
+                    "global_target_fpr",
+                    "(Fairness) Global target FPR",
+                    "This is an inequality constraint: inactive when FPR is lower than the target. "
```

Comment:
Start with: "(Fairness) Global constraint type must select FPR for this value to matter."

Do the same for the FNR below.